src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets

```diff
@@ -10,7 +10,8 @@
     <OSIdentifier Condition="'$(OSIdentifier)' == ''">linux</OSIdentifier>
 
     <RuntimeIlcPackageName>runtime.$(OSIdentifier)-$(PlatformTarget).Microsoft.DotNet.ILCompiler</RuntimeIlcPackageName>
-    <IlcHostArch Condition="'$(IlcHostArch)' == ''">$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant)</IlcHostArch>
+    <IlcHostArch Condition="'$(IlcHostArch)' == '' and ('$(PROCESSOR_ARCHITECTURE)' == 'ARM64' or '$(PROCESSOR_ARCHITEW6432)' == 'ARM64')">arm64</IlcHostArch>
```

Comment:
```suggestion
    <!-- Cannot use RuntimeInformation.ProcessArchitecture because that's the architecture of the process running MSBuild; assume ARM64 on ARM64 and x64 otherwise. -->
    <IlcHostArch Condition="'$(IlcHostArch)' == '' and ('$(PROCESSOR_ARCHITECTURE)' == 'ARM64' or '$(PROCESSOR_ARCHITEW6432)' == 'ARM64')">arm64</IlcHostArch>
```

Reply:
That is sort of obvious in retrospect.  I actually intended to use `RuntimeInformation.OSArchitecture` instead of `ProcessArchitecture`. The former would work for x86 and x64 processes running on x64; however, it would lie for x86 and x64 processes running on ARM64.  Unfortunately, the `PROCESSOR_*` environment variables also lie in case of an x64 process running on ARM64 with recent Windows preview builds.  However, in that case we are at least able to run the x64-hosted ILCompiler, it would be just slower.  As far as I can tell, reading from the registry
```PowerShell
Get-ItemPropertyValue 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment' 'PROCESSOR_ARCHITECTURE'
```
never lies; however, the .NET Core version of MSBuild comes with no registry provider, so we would have to read it by other means, e.g., `reg query`.  That is why I left the option to override `IlcHostArch` if needed.  We will need to review this later.